test/bufconn/bufconn_test.go

```diff
@@ -197,3 +197,103 @@ func TestCloseWhileAccepting(t *testing.T) {
 		t.Fatalf("c, err = %v, %v; want nil, %v", c, err, errClosed)
 	}
 }
+
+func TestDeadline(t *testing.T) {
+	sig := make(chan struct{})
+	var blockingWrite = func(conn net.Conn) {
+		conn.Write([]byte("0123456789"))
+		sig <- struct{}{}
```

Comment:
The return values of `Read`/`Write` is never checked.  How about making `sig` a `chan error` and passing the errors returned by `Read`/`Write` on it, to be verified in the test body?